### PR TITLE
[TIMOB-25020] Android: requestThumbnailImagesAtTimes causes a crash

### PR DESCRIPTION
**Pull request by ypbnv**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-25020

**Description:**
Passing a URI with 'file' scheme causes the ContentResolver's cursor to throw an error.

Handle file resources URI manually.
Change setDataSource method to work with all resource types.

Note: I am having some problem with running embedded videos after the 7.0.0 update and this PR's changes. It may be a device specific (HTC One Max), because it is a native error and I can't get any meaningful info. Test case for a potential regression can be found here:
https://github.com/appcelerator/titanium_mobile/pull/8830

### CODE REVIEW

File: android/modules/media/src/java/ti/modules/titanium/media/TiThumbnailRetriever.java [hunk 1]
```diff
@@ -171,8 +171,7 @@ private int setDataSource(Uri mUri, MediaMetadataRetriever mMediaMetadataRetriev
 					} else {
 						mUri = TiUIHelper.getRedirectUri(mUri);
 						if (Build.VERSION.SDK_INT >= 14){
-							FileInputStream inputStream = new FileInputStream(mUri.getPath());
-							mMediaMetadataRetriever.setDataSource(inputStream.getFD());
+							mMediaMetadataRetriever.setDataSource(TiApplication.getAppRootOrCurrentActivity(),mUri);
```
Comment (garymathews) [inline android/modules/media/src/java/ti/modules/titanium/media/TiThumbnailRetriever.java]: formatting, space after `,`

**Comment by ypbnv** [comment]
@garymathews Fixed formatting. Removed some unused imports.

File: android/modules/media/src/java/ti/modules/titanium/media/VideoPlayerProxy.java [hunk 2]
```diff
@@ -718,13 +718,18 @@ public void onDestroy(Activity activity)
...
 		if (hasProperty(TiC.PROPERTY_URL)) {
-			String url = TiConvert.toString(getProperty(TiC.PROPERTY_URL));
-			String path = url.contains(":") ? new TitaniumBlob(url).getNativePath() : resolveUrl(null, url);
-			Uri uri = Uri.parse(path);
-
 			cancelAllThumbnailImageRequests();
 			mTiThumbnailRetriever = new TiThumbnailRetriever();
-			mTiThumbnailRetriever.setUri(uri);
+			String url = TiConvert.toString(getProperty(TiC.PROPERTY_URL));
+			if (url.contains("file://")) {
```
Comment (garymathews) [inline android/modules/media/src/java/ti/modules/titanium/media/VideoPlayerProxy.java]:
```Java
url.startsWith("file://")
```

**Review by garymathews — CHANGES_REQUESTED** [review]
See above

**Comment by ypbnv** [comment]
@garymathews Updated.

**Review by garymathews — APPROVED** [review]
CR: **PASS**

**Review by ssjsamir — APPROVED** [review]
FR Passed: When using `requestThumbnailImagesAtTimes` application no longer crashes. 


**Test Steps**
+ Downloaded the SDK Build from this PR
+ Created a new Titanium project
+ Used the following test case:
```
var win = Ti.UI.createWindow({
		title: 'test',
		backgroundColor: 'white',
		layout: 'vertical'
});
	
var button = Ti.UI.createButton({
    title: "record"
});
 
button.addEventListener('click', recordVideo);
 
win.add(button);
win.open();
 
function recordVideo() {
    var permissionsToRequest = [];
 
   
 
    var cameraPermission = "android.permission.CAMERA";
    var hasCameraPerm = Ti.Android.hasPermission(cameraPermission);
    if (!hasCameraPerm) {
        permissionsToRequest.push(cameraPermission);
    }
 
    var storagePermission = "android.permission.READ_EXTERNAL_STORAGE";
    var hasStoragePerm = Ti.Android.hasPermission(storagePermission);
    if (!hasStoragePerm) {
        permissionsToRequest.push(storagePermission);
    }
    if(permissionsToRequest.length > 0) {
        Ti.Android.requestPermissions(permissionsToRequest, function (e) {
            if (e.success) {
                showCamera();
            }
        });
    } else {
        showCamera();
    }
}
 
function showCamera() {
    Ti.Media.showCamera({
        autohide: false,
        animated: false,
        allowEditing: false,
        success:function(event) {
            var movie = Titanium.Media.createVideoPlayer({
                url:event.media.nativePath,
                backgroundColor:'#111',
                mediaControlStyle: Titanium.Media.VIDEO_CONTROL_EMBEDDED,
                scalingMode:Titanium.Media.VIDEO_SCALING_ASPECT_FILL,
                width:300,
                height:300,
            });
            win.add(movie);
 
            movie.addEventListener('durationavailable', function() {
                movie.requestThumbnailImagesAtTimes([1], Titanium.Media.VIDEO_TIME_OPTION_NEAREST_KEYFRAME, function(response) {
                    Ti.API.info("Thumbnail callback called, success = " + response.success);
                    console.log(response);
                    if(response.success) {
 
                    }
                });
            });
        },
        cancel:function() {
            console.log("cancel");
        },
        error:function(error) {
            console.log("error");
            console.log(error);
        },
        mediaTypes: [Ti.Media.MEDIA_TYPE_VIDEO],
    });
}
```
+ Ran the application
+ Pressed record
+ Recorded a short video
+ Pressed okay
+ Was able to see the thumbnail from the previous menu 

**Test Environment**
Appcelerator Command-Line Interface, version 6.2.3
Google Nexus 6P (7.1.1)
Operating System Name: Mac OS X El Capitan
Operating System Version: 10.11.6
Node.js Version: 6.10.1
Xcode: 8.2
Appcelerator Studio: 4.9.0.201705302345

**Pull request merged**